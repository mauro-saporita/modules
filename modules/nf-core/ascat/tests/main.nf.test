nextflow_process {

    name "Test Process ASCAT"
    script "../main.nf"
    process "ASCAT"

    tag "modules"
    tag "modules_nfcore"
    tag "ascat"

    test("ascat") {

        config "./nextflow.config"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test', single_end:false ],
                    [
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_sorted_bam'], checkIfExists: true),
                        file(params.test_data['homo_sapiens']['illumina']['test2_paired_end_sorted_bam_bai'], checkIfExists: true)
                    ]
                ]
                input[1] = []
                input[2] = []
                input[3] = []
                input[4] = []
                input[5] = []
                input[6] = []
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out.versions).match("versions") }
            )
        }
    }
}