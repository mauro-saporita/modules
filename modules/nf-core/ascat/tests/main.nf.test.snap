{
    "human - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.normal_alleleFrequencies_chr21.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.normal_alleleFrequencies_chr22.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_alleleFrequencies_chr21.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_alleleFrequencies_chr22.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.tumour_normalBAF.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_tumourBAF.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cnvs.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.tumour_normalLogR.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_tumourLogR.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.metrics.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.after_correction.gc_rt.test.tumour.germline.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.after_correction.gc_rt.test.tumour.tumour.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.before_correction.test.tumour.germline.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.before_correction.test.tumour.tumour.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour.ASPCF.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour.sunrise.png:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.purityploidy.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "7": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.segments.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "8": [
                    "versions.yml:md5,257953220f0dc2fc0bc978cb5c26cc62"
                ],
                "allelefreqs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.normal_alleleFrequencies_chr21.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.normal_alleleFrequencies_chr22.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_alleleFrequencies_chr21.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_alleleFrequencies_chr22.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "bafs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.tumour_normalBAF.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_tumourBAF.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "cnvs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.cnvs.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "logrs": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.tumour_normalLogR.txt:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour_tumourLogR.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "metrics": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.metrics.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "png": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        [
                            "test.after_correction.gc_rt.test.tumour.germline.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.after_correction.gc_rt.test.tumour.tumour.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.before_correction.test.tumour.germline.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.before_correction.test.tumour.tumour.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour.ASPCF.png:md5,f50b84b1db4b83ba62ec1deacc69c260",
                            "test.tumour.sunrise.png:md5,f50b84b1db4b83ba62ec1deacc69c260"
                        ]
                    ]
                ],
                "purityploidy": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.purityploidy.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "segments": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.segments.txt:md5,f50b84b1db4b83ba62ec1deacc69c260"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,257953220f0dc2fc0bc978cb5c26cc62"
                ]
            }
        ],
        "timestamp": "2024-03-21T12:43:33.378606"
    },
    "versions": {
        "content": [
            [
                
            ]
        ],
        "timestamp": "2024-03-21T12:43:29.300939"
    }
}